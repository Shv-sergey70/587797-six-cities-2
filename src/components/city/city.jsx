import React from 'react';
import PropTypes from 'prop-types';

export const City = (props) => {
  const {
    cityName,
    isActive,
    onCityClick
  } = props;

  return <li className="locations__item">
    <a className={`locations__item-link tabs__item ${isActive ? `tabs__item--active` : ``}`} href="#" onClick={(evt) => {
      onCityClick(evt.target.innerText);
    }}>
      <span>{cityName}</span>
    </a>
  </li>;
};

City.propTypes = {
  cityName: PropTypes.string,
  isActive: PropTypes.bool,
  onCityClick: PropTypes.func.isRequired
};
