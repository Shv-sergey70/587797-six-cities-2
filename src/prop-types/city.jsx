import PropTypes from "prop-types";
import locationPropTypes from './location';

export default PropTypes.exact({
  name: PropTypes.string.isRequired,
  location: locationPropTypes
}).isRequired;
