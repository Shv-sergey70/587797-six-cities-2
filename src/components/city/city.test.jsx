import React from 'react';
import renderer from 'react-test-renderer';
import {City} from "./city";

const cityData = {
  name: `Paris`,
  location: {
    latitude: 11,
    longitude: 22,
    zoom: 10
  }
};

describe(`City correctly renders`, () => {
  it(`City is active`, () => {
    const tree = renderer.create(
        <City
          city = {cityData}
          currentCity = {{
            name: `Paris`,
            location: {
              latitude: 11,
              longitude: 22,
              zoom: 10
            }
          }}
          changeCity = {jest.fn()}
        />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`City is not active`, () => {
    const tree = renderer.create(
        <City
          city = {cityData}
          currentCity = {{
            name: `Amsterdam`,
            location: {
              latitude: 11,
              longitude: 22,
              zoom: 10
            }
          }}
          changeCity = {jest.fn()}
        />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
