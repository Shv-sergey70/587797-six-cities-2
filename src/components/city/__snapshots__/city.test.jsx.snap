// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`City correctly renders City is active 1`] = `
<li
  className="locations__item"
>
  <a
    className="locations__item-link tabs__item tabs__item--active"
    href="#"
    onClick={[Function]}
  >
    <span>
      Paris
    </span>
  </a>
</li>
`;

exports[`City correctly renders City is not active 1`] = `
<li
  className="locations__item"
>
  <a
    className="locations__item-link tabs__item "
    href="#"
    onClick={[Function]}
  >
    <span>
      Paris
    </span>
  </a>
</li>
`;
